src/app/record/components/funding-stacks-groups/funding-stacks-groups.component.ts

```diff
@@ -40,16 +41,26 @@ export class FundingStacksGroupsComponent implements OnInit {
   ngOrcidFunding = $localize`:@@shared.funding:Funding`
   countryCodes: { key: string; value: string }[]
   loading = true
+  sourceSorting = false
+  sortTypes: SortOrderType[] = ['title', 'start', 'end']
 
   constructor(
     private _userSession: UserService,
     private _record: RecordService,
     private _recordFundingsService: RecordFundingsService,
-    private _recordCountryService: RecordCountriesService
+    private _recordCountryService: RecordCountriesService,
+    private _togglz: TogglzService
   ) {}
 
   ngOnInit(): void {
+    this._togglz
+       .getStateOf('SOURCE_SORTING')
+       .pipe(take(1))
+       .subscribe((value) => (this.sourceSorting = value))
     this.getRecord()
+    if (this.sourceSorting) {
```

Comment:
Since this `if (this.sourceSorting)` is out the subscription scope. It might have a unstable behaviour. Since the subscription might or might not have assigned the final value for `sourceSorting`. 

I think we can just move the if inside the subscription, so we now for sure what is the flag status.

Reply:
Great catch! You are right this togglz should be handled differently than the others. I will make the changes. Thanks